### ISSUE TITLE
2B model

### ISSUE THREAD

**Issue by KooSung** [raghavlite/B3#1]
Nice work! I was wondering if there are any plans to release the 2B model? If so, is there an estimated timeline for when it might be made available?

**Comment by raghavlite** [comment]
Hi. 
The 2B model is released.

**Comment by KooSung** [comment]
@raghavlite Thanks! Is there a script for extracting embeddings?

**Comment by raghavlite** [comment]
Hi,
The **eval_mmeb.py** is fairly simple. Creates a data loader and iterates through a dataset. You should be able to repurpose your data into that format.

The exact extraction of embeddings happens here in the **eval_mmeb.py** script.
https://github.com/raghavlite/B3/blob/d2773f1d37a44e25e31e2ce8f0370af413370048/eval_mmeb.py#L160

You can use this script for now.


I'm currently busy for a couple of days. I'll create a more clear script (where you can supply an image and a text) and add it to the readme by Sunday.

**Comment by zhangyongshun** [comment]
> Hi, The **eval_mmeb.py** is fairly simple. Creates a data loader and iterates through a dataset. You should be able to repurpose your data into that format.
> 
> The exact extraction of embeddings happens here in the **eval_mmeb.py** script.
> 
> [B3/eval_mmeb.py](https://github.com/raghavlite/B3/blob/d2773f1d37a44e25e31e2ce8f0370af413370048/eval_mmeb.py#L160)
> 
> Line 160 in [d2773f1](/raghavlite/B3/commit/d2773f1d37a44e25e31e2ce8f0370af413370048)
> 
>  encoded_tensor.append(output["qry_reps"].cpu().detach().float()) 
> You can use this script for now.
> 
> I'm currently busy for a couple of days. I'll create a more clear script (where you can supply an image and a text) and add it to the readme by Sunday.

any update?

**Comment by raghavlite** [comment]
Sorry. I've been swamped. Will definitely do it this weekend. 
Give it a try yourself too. It should be very easy.
